lib/helpers/model/discriminator.js

```diff
@@ -116,6 +116,7 @@ module.exports = function discriminator(model, name, schema, tiedValue, applyPlu
     }
 
     mergeDiscriminatorSchema(schema, baseSchema);
+    schema.tree = Object.assign({}, baseSchema.tree, schema.tree);
```

Comment:
why is this call outside of `mergeDiscriminatorSchema` instead of inside as a separate case?

Reply:
You're right, that's better design. Will fix.